### DISCUSSION TITLE
Deploy Cruise Control to an existing cluster ?

### DISCUSSION THREAD

**Discussion by chanvoyage** [strimzi/strimzi-kafka-operator#5144] [Q&A]
hello guy

Is possible Deploy  Cruise Control to an existing cluster ?

**Comment by scholzj** [comment]
You can just edit your Kafka custom resource and add Cruise Control to it: https://strimzi.io/docs/operators/latest/full/using.html#proc-deploying-cruise-control-str ... the operator will take care of the rest (Note: this will do a rolling update of the Kafka brokers to install the Cruise Control metrics reporter)

**Reply by chanvoyage** [reply]
thank you  scholzi, I have some question
 i  was successful deploy kafka  connect cluster and service for kafka connect 
i am wondering when  get service of kafka connect and kafka zookeeper  are  Cluster IP  , this mean can not connect out side cluster , i try to create node port it does not work 
Do you have anything suggestion ?

![image](https://user-images.githubusercontent.com/39899497/122510385-b003a000-d02f-11eb-9e62-b65e2f7bc48c.png)

![image](https://user-images.githubusercontent.com/39899497/122510452-d590a980-d02f-11eb-8532-61e9879613c6.png)

**Reply by scholzj** [reply]
Strimzi does not expose ZooKeeper. Historically, it is hard to secure and it will be anyway removed from Kafka fairly soon. So even if you create an additional service, you will fail the TLS authentication. If you really need to access ZooKeeper, you can at your own risk use this unofficial project: https://github.com/scholzj/zoo-entrance. But best is to avoid it.

As for Kafka Connect, you can create your own service if you want to access it from the outside - again, at your own risk as it will not be secured etc.

**Reply by chanvoyage** [reply]
Dear scholzj

thank you for information  ,

As for Kafka Connect, you can create your own service if you want to access it from the outside - again, at your own risk as it will not be secured etc.

ans:   this mean  kafka connect can not access out side cluster because service is Cluster IP ,  Do you have suggestion how  to connect  database sql server and oracle bare metal  or anything else  ?


Is possible create   Kafka connect   with  nodeport or ingress ? 

i test to create nodeport  with selector of kafka connect pods ...    and test  access it does not work 

Could you please any suggestion  for kafka connect. can connect out side cluster ? 

thank you very much 

Note:  I use  azure aks  for deploy kafka

**Reply by scholzj** [reply]
> this mean kafka connect can not access out side cluster because service is Cluster IP , Do you have suggestion how to connect database sql server and oracle bare metal or anything else ?

No, that is wrong. The service is only about you accessing the Kafka Connect REST API. It has no relation to Kafka Connect connecting to some DB.

> i test to create nodeport with selector of kafka connect pods ... and test access it does not work

If you enabled the connector operator in the Kafka Connect deployment, you might need to add a network policy for that as well. With nodeport, you might also need to set some firewall rules etc. in your Azure account.
